libs/community/langchain_community/vectorstores/falkordb_vector.py

```diff
@@ -0,0 +1,1684 @@
+        if index_type == "NODE":
+            raise ValueError(
+                "Node vector index is not supported with "
+                "`from_existing_relationship_index` method. Please use the "
+                "`from_existing_index` method."
+            )
+
+        # Check if embedding function and vector index dimensions match
+        if embedding_dimension and not store.embedding_dimension == embedding_dimension:
+            raise ValueError(
+                "The provided embedding function and vector index "
+                "dimensions do not match.\n"
+                f"Embedding function dimension: {store.embedding_dimension}\n"
+                f"Vector index dimension: {embedding_dimension}"
+            )
+
+        return store
+    
+    
+    def add_documents(
+        self,
+        documents: List[Document],
+    ) -> List:
+        """
+        This function takes List[Document] element(s) and populates
+        the existing store with a default node or default node(s) that 
+        represents the element(s) and returns the id(s) of the newly created node(s)
+
+        Args:
+            documents: the List[Document] element
+        Returns:
+            a list containing the id(s) of the newly created node in the store
+        """
+        ids = []
+
+        #set the index type
+        self._index_type = 'NODE'
+
+        # Add the documents to the store
+        self.from_documents(
+            embedding=self.embedding,
+            documents=documents,
+        )
+
+        for doc in documents:
+            page_content = doc.page_content
+            result = self.query(
+                """
+                MATCH (n)
+                WHERE n.text = $page_content
+                RETURN n.id
+                """, 
+                params={"page_content": page_content}
+            )
+            try:
+                ids.append(result[0][0])
+            except Exception as e:
+                raise ValueError(
+                    "Your document wasn't added to the store successfully check your spellings"
+                )
+
+        return ids
+            
+
+
+
+        
+        
+
+    @classmethod
+    def from_documents(
+        cls: Type[FalkorDBVector],
+        documents: List[Document],
+        embedding: Embeddings,
+        distance_strategy: DistanceStrategy = DEFAULT_DISTANCE_STRATEGY,
+        ids: Optional[List[str]] = None,
+        **kwargs: Any,
+    ) -> FalkorDBVector:
+        """
+        Return FalkorDBVector initialized from documents and embeddings.
+        """
+        texts = [d.page_content for d in documents]
+        metadatas = [d.metadata for d in documents]
+        
+
+        return cls.from_texts(
+            texts=texts,
+            embedding=embedding,
+            distance_strategy=distance_strategy,
+            metadatas=metadatas,
+            ids=ids,
+            **kwargs,
+        )
+    
+    @classmethod
+    def from_existing_graph(
```

Comment:
How does one specifies which Graph to connect to?

Reply:
It didn't, thank you for bringing that to my attention. I've added a new parameter to the function `database`. See;
```
@classmethod
    def from_existing_graph(
        cls: Type[FalkorDBVector],
        embedding: Embeddings,
        database: str,
        node_label: str,
        embedding_node_property: str,
        text_node_properties: List[str],
        *,
        search_type: SearchType = DEFAULT_SEARCH_TYPE,
        retrieval_query: str = "",
        **kwargs: Any,
    ) -> FalkorDBVector:
```

Reply:
I added the parameter `database` to `from_existing_graph` to tackle this.